docs/site/migration/models/methods.md

```diff
@@ -6,8 +6,309 @@ sidebar: lb4_sidebar
+  })
+  async count(
+    @param.query.object('where', getWhereSchemaFor(Note)) where?: Where<Note>,
+  ): Promise<Count> {
+    return this.noteRepository.count(where);
+  }
+
+  @get('/notes', {
+    responses: {
+      '200': {
+        description: 'Array of Note model instances',
+        content: {
+          'application/json': {
+            schema: {
+              type: 'array',
+              items: getModelSchemaRef(Note, {includeRelations: true}),
+            },
+          },
+        },
+      },
+    },
+  })
+  async find(
+    @param.query.object('filter', getFilterSchemaFor(Note)) filter?: Filter<Note>,
+  ): Promise<Note[]> {
+    return this.noteRepository.find(filter);
+  }
+
+  @patch('/notes', {
+    responses: {
+      '200': {
+        description: 'Note PATCH success count',
+        content: {'application/json': {schema: CountSchema}},
+      },
+    },
+  })
+  async updateAll(
+    @requestBody({
+      content: {
+        'application/json': {
+          schema: getModelSchemaRef(Note, {partial: true}),
+        },
+      },
+    })
+    note: Note,
+    @param.query.object('where', getWhereSchemaFor(Note)) where?: Where<Note>,
+  ): Promise<Count> {
+    return this.noteRepository.updateAll(note, where);
+  }
+
+  @get('/notes/{id}', {
+    responses: {
+      '200': {
+        description: 'Note model instance',
+        content: {
+          'application/json': {
+            schema: getModelSchemaRef(Note, {includeRelations: true}),
+          },
+        },
+      },
+    },
+  })
+  async findById(
+    @param.path.number('id') id: number,
+    @param.query.object('filter', getFilterSchemaFor(Note)) filter?: Filter<Note>
+  ): Promise<Note> {
+    return this.noteRepository.findById(id, filter);
+  }
+
+  @patch('/notes/{id}', {
+    responses: {
+      '204': {
+        description: 'Note PATCH success',
+      },
+    },
+  })
+  async updateById(
+    @param.path.number('id') id: number,
+    @requestBody({
+      content: {
+        'application/json': {
+          schema: getModelSchemaRef(Note, {partial: true}),
+        },
+      },
+    })
+    note: Note,
+  ): Promise<void> {
+    await this.noteRepository.updateById(id, note);
+  }
+
+  @put('/notes/{id}', {
+    responses: {
+      '204': {
+        description: 'Note PUT success',
+      },
+    },
+  })
+  async replaceById(
+    @param.path.number('id') id: number,
+    @requestBody() note: Note,
+  ): Promise<void> {
+    await this.noteRepository.replaceById(id, note);
+  }
+
+  @del('/notes/{id}', {
+    responses: {
+      '204': {
+        description: 'Note DELETE success',
+      },
+    },
+  })
+  async deleteById(@param.path.number('id') id: number): Promise<void> {
+    await this.noteRepository.deleteById(id);
+  }
+}
+
+```
+
+To hide a particular REST API endpoint, simply delete the appropriate function and its decorator.
+
+To hide all REST API endpoints of a particular model, simply do not create a controller class for it.
+
+
```

Comment:
Can you please explain how to customize a particular REST API endpoint? How to add a new (custom) endpoint?